### ISSUE TITLE
Can't add content to "Content" area of a screen or form

### ISSUE THREAD

**Issue by jeffslofish** [FelixHenninger/lab.js#7]
According to the docs, it looks like you should be able to add HTML to the Content area. There is no apparent way to add anything to this area. This is happening both on my local build and on https://labjs.felixhenninger.com/. I would have expected that when I click on the Content area, I get a cursor and can start typing, but no such thing happens.

**Comment by jeffslofish** [comment]
I looked into it and it turns out to be due to a Javascript error caused by the following

````
      url: process.env.PUBLIC_URL + '/vendor/require.js',
````
on line 35 in Editor/index.js.

/ventor/require.js is actually and HTML file and so it tries to parse it as js and gives an error and prevent further processing.

**Comment by FelixHenninger** [comment]
Hi Jeff,

thanks for stopping by and taking the time to give feedback, and especially for starting to debug the issue! It's great to have an entry for this, because the symptoms you are describing have been recurring, and I would really like to solve them. This is definitely a problem on my side!

If I understand you correctly, what you're describing is that the editor is not starting. This seems to happen if the editor's source files are not loaded in time. As a quick fix, does re-loading the entire UI (possibly doing a [force refresh](http://refreshyourcache.com/en/cache/) by pressing the `shift` key) change anything?
As far as I understand, our editor wants to load its own dependencies (via `require.js`), and therefore its files are not bundled with the bulk of the application. That's why they are not loaded immediately. Being fairly bulky, it looks like they trigger some time-out later, which causes the editor to fail (this part is my speculation).

An HTML file being in place of `require.js` is telling, because it points to another issue: I recently started making the app [progressive](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#making-a-progressive-web-app), so that all assets are cached in the local browser. In principle, this should make the builder offline-capable, but it seems to introduce new issues in combination with the one above: The client-side cache (which is not aware of the editor's dependencies) seems to substitute the editor source with an invalid file.

Here's what I'll try: I'm going to see if the editor source can be preloaded with the page, and included in the cache -- I hope this is going to fix both of the above problems.

Ok, so much for now -- I'll look into this tonight and over the coming days, and very much hope that we can solve this. Sorry for the trouble and annoyance this has caused, and thanks again for your helpful and thoughtful report! If there's anything I can do to help, or any questions I could answer, please do let me know!

All the very best,

-Felix

**Comment by FelixHenninger** [comment]
Just a quick update: I can replicate the issue by throttling the network connection. The editor loads for me once the files are cached, but the first attempt always fails if I reduce the speed. I probably hadn't noticed this before because I'm very close to the server, and well-connected.

With regard to the second issue, the UI loading regardless of the path (even for URLs that should show JavaScript code, like `vendor/require.js`) seems to be a 'feature' of the preview worker. I'm not 100% sure, but it looks to me like this behavior is limited to the browser UI. If I try to access the same data through a script, it shows that the file is actually in place:

```
fetch('/vendor/require.js')
  .then(response => response.text())
  .then(content => console.log(content))
# ... prints file contents
```

**Comment by FelixHenninger** [comment]
This is addressed and (hopefully) fixed in dc5d7da845ecba7b23e0e2404f641159105dac30, which causes all assets to be preloaded as suggested above, including the editor code files.

Huge thanks are due to @jeffslofish for reporting this &mdash; we have been in contact and Jeff has reported that the issue is solved by the commit.
